src/lib/utils.jl

```diff
@@ -111,3 +111,29 @@ a real.
      1  0
 """
 hessian(f, x::AbstractArray) = forward_jacobian(x -> gradient(f, x)[1], x)[2]
+
+"""
+   isderiving()
+   isderiving(x)
+
+Check whether the current function call is happening while taking the derivative.
```

Comment:
We should specify what is the argument `x`

Reply:
This is common practice if the input can be generic - adding an `Any` adds no value

Reply:
I'm not talking about the type, I mean when one should use `isderiving()` and when `isderiving(x)`, the docstring only mentions a current function call without any reference to x